tests/rptest/tests/tx_admin_api_test.py

```diff
@@ -244,3 +246,216 @@ def test_all_transactions(self):
                 expected_partitions[topic.name]) == topic.partition_count
             for partition in range(topic.partition_count):
                 assert partition in expected_partitions[topic.name]
+
+    @cluster(num_nodes=3)
+    def test_delete_topic_from_ongoin_tx(self):
+        tx_id = "0"
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+
+        for topic in self.topics:
+            for partition in range(topic.partition_count):
+                producer.produce(topic.name, '0', '0', partition)
+
+        producer.flush()
+
+        txs_info = self.admin.get_all_transactions()
+        assert len(
+            txs_info) == 1, "Should be only one transaction in current time"
+
+        rpk = RpkTool(self.redpanda)
+        topic_name = self.topics[0].name
+        rpk.delete_topic(topic_name)
+
+        tx = txs_info[0]
+        assert tx[
+            "transactional_id"] == tx_id, f"Expected transactional_id: {tx_id}, but got {tx['transactional_id']}"
+
+        for partition in tx["partitions"]:
+            assert (partition["ns"] == "kafka")
+            if partition["topic"] == topic_name:
+                self.admin.delete_partition_from_transaction(
+                    tx["transactional_id"], partition["ns"],
+                    partition["topic"], partition["partition_id"],
+                    partition["etag"])
+
+        producer.commit_transaction()
+
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+
+        for topic in self.topics:
+            if topic.name is not topic_name:
+                for partition in range(topic.partition_count):
+                    producer.produce(topic.name, '0', '0', partition)
+
+        producer.commit_transaction()
+
+    @cluster(num_nodes=3)
+    def test_delete_topic_from_prepared_tx(self):
+        tx_id = "0"
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+
+        for topic in self.topics:
+            for partition in range(topic.partition_count):
+                producer.produce(topic.name, '0', '0', partition)
+
+        producer.flush()
+
+        rpk = RpkTool(self.redpanda)
+        topic_name = self.topics[0].name
+        rpk.delete_topic(topic_name)
+
+        try:
+            producer.commit_transaction()
```

Comment:
can you add an assert after commit_transaction - now we check that an exception maybe thrown while we should check that an exception is thrown